service/history/workflow/mutable_state_impl.go

```diff
@@ -428,6 +428,8 @@ func (e *MutableStateImpl) getCurrentBranchTokenAndEventVersion(eventID int64) (
 // SetHistoryTree set treeID/historyBranches
 func (e *MutableStateImpl) SetHistoryTree(
 	ctx context.Context,
+	executionTimeout *time.Duration,
+	runTimeout *time.Duration,
```

Comment:
It looks like ideally we will want those information available when creating the mutable state so that we have to pass in them here. 

If those are remove right now, then we can only rely on caller to add started event before setting the history tree, which is bad I think. So I think explicitly passing in the two timeouts does make sense. Maybe add a TODO comment here saying explore the possibility of providing timeouts in mutable state constructor?